Fix: DOI serach validation and error display

lpandath opened the pull request:
<!-- Please make sure you have read the submission guidelines before posting an PR -->
<!-- https://github.com/tuwien-csd/damap-backend/blob/next/CONTRIBUTING.md#pullrequests -->

## Description

Add immediate validation when user enters a DOI
Show error messages right away when DOI is invalid
Prevent search with invalid DOIs
Update button text 
Use UntypedFormControl to match existing codebase patterns


<img width="756" alt="Screenshot 2025-05-30 at 11 59 17" src="https://github.com/user-attachments/assets/cffc215c-38fb-47d5-b09e-31b3cdefe17a" />

<img width="765" alt="Screenshot 2025-05-30 at 11 58 57" src="https://github.com/user-attachments/assets/9448051b-7db2-4c40-8e9f-13e6b30dca41" />


Test with DOI: 10.5281/zenodo.3885730

#### What does this PR do?

<!-- Changes introduced by this PR - what happened before, what happens now -->

#### Breaking changes

<!-- Whether this PR contains breaking changes and which -->

#### Code review focus

<!-- What you want the reviewer to focus on -->

#### Dependencies

<!-- If this PR depends on or requires other/additional (backend) changes, please list them here -->

### Checks

<!-- Adjust list as necessary -->

- [ ] Summary updated
- [ ] Version view updated
- [ ] Documentation added
- [x] Tests added
- [ ] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-<!-- insert issue number -->

lpandath committed:
Fix: DOI serach validation and error display

ValentinFutterer commented:
Solid PR. I have only one thing to note - its still possible to search with an empty search field. In this case, the "Add Dataset" button should be blocked.

lpandath committed:
fix: disable DOI search button when field is empty

- add validator to DOI form control
- empty DOI searches by blocking Add Dataset button
- combine required and DOI format validation

ValentinFutterer commented:
Thank you, can you squash the two commits, then I will merge 👍
